mreg_cli/api/models.py

```diff
@@ -454,13 +470,14 @@ class Role(FrozenModelWithTimestamps, APIMixin["Role"]):
 
     id: int  # noqa: A003
     created_at: datetime = Field(..., validation_alias=AliasChoices("create_date", "created_at"))
-    hosts: List[str]
-    atoms: List[str]
+    hosts: NameList
+    atoms: NameList
```

Comment:
You can decide if you want to use `List[Annotated[str, BeforeValidator(_extract_name)]]` for each field or keep using `NameList`. I suppose keeping the `NameList` type around could be useful if we have more such fields?

Reply:
NameList can also offer some more stuff like outputting the values etc.